src/stateMachine.js

```diff
@@ -81,15 +83,24 @@ function updateEntity (keyPath, ts) {
  * @param {Window} window The window object.
  * @param {Object} state The current state.
  * @param {Object} event The event.
+ * @param {Object} emit The emitter object.
  * @returns A new state.
  */
-function reducer (promos, window, state, event) {
+const stateMachine = (promos, window) => ({ user }, event, emit) => {
   const ts = timestamp()
-  let { user } = state
 
   if (event.type === 'visit') {
     // Increment the number of user visits.
     user = incrementVisits(user)
+
+    // Place the promos.
+    const placedPromos = placePromos(promos, user, window)
+
+    // Track the impressions for the placed promos.
+    user = trackImpressions(placedPromos, ts, user)
+
+    // Emit the placed promos.
+    emit.next({ promos: placedPromos })
```

Comment:
This is where the promos are actually emitted. i.e. They will only be emitted on the initial `visit` event. Any other events will **not** trigger the promos to be placed again (although they will update the internal user state object).